src/app/leverage/provider.tsx

```diff
@@ -443,7 +444,8 @@ export function LeverageProvider(props: { children: any }) {
       !!address &&
       !!chainId &&
       !!publicClient &&
-      !!indexToken &&
+      !!inputToken &&
+      !!outputToken &&
```

Comment:
was just depending on the `indexToken` which doesn't change if `outputToken` is a currency.